mysql-test/suite/tokudb/t/in_backup.test

```diff
@@ -0,0 +1,266 @@
+INSERT INTO test.t2 VALUES(1122,1122);
+
+# do checkpoint
+SET GLOBAL tokudb_checkpoint_on_flush_logs=ON;
+FLUSH LOGS;
+SET GLOBAL tokudb_checkpoint_on_flush_logs=OFF;
+
+connect (conn1, localhost, root,,);
+connection conn1;
+
+#conn1 write(1,1,1)
+INSERT INTO test.t1 VALUES(1,1,1);
+# fake xtrabackup 1) tokudb checkpoint lock
+SET GLOBAL TOKUDB_CHECKPOINT_LOCK=ON;
+
+connect (conn2, localhost, root,,);
+connection conn2;
+
+# table for drop during backup
+CREATE TABLE test.t22 (c1 INT PRIMARY KEY, c2 INT) ENGINE=TOKUDB;
+INSERT INTO test.t22 VALUES(1,1);
+
+# conn2 write(2,2,2)
+INSERT INTO test.t1 VALUES(2,2,2);
+
+connection conn1;
+# fake xtrabackup 2) FLUSH TABLES WITH READ LOCK
+FLUSH TABLES WITH READ LOCK;
+
+# fake xtrabackup 3) copy tokudb.* and redo-log to backup dir
+--exec cp -rf $MYSQLD_DATADIR/tokudb.* $MYSQL_TMP_DIR/backupdir/
+--exec cp -rf $MYSQLD_DATADIR/log000* $MYSQL_TMP_DIR/backupdir/
+# fake extrabackup 4) unlock tables
+UNLOCK TABLES;
+
+connection conn2;
+# shadow row
+--echo conn2 write(22,22,22), this row can't visible in new backup instance
+INSERT INTO test.t1 VALUES(22222,22222,22222);
+FLUSH TABLES;
+
+connection conn1;
+# fake xtrabackup 5) copy tokudb datas file to backup dir
+ALTER TABLE test.t11 RENAME test.t111;
+--exec cp -rf $MYSQLD_DATADIR/_test_t1* $MYSQL_TMP_DIR/backupdir/
+# fake drop tables during backup
+DROP TABLE test.t22;
+--exec cp -rf $MYSQLD_DATADIR/_test_t2* $MYSQL_TMP_DIR/backupdir/
+DROP TABLE test.t111;
+
+# fake xtrabackup 6) release tokudb checkpoint lock
+SET GLOBAL TOKUDB_CHECKPOINT_LOCK=OFF;
+
+--echo shutdown server
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--shutdown_server 10
+--source include/wait_until_disconnected.inc
+
+# cleanup tokudb files
+--exec rm -rf $MYSQLD_DATADIR/tokudb.*
+--exec rm -rf $MYSQLD_DATADIR/log000*
+
+# fake recovery: copy tokudb.* and redo-log, data files from backup dir
+--exec cp -rf $MYSQL_TMP_DIR/backupdir/* $MYSQLD_DATADIR/
+
+--enable_reconnect
+--exec echo "restart: --innodb_buffer_pool_size=100M --tokudb_cache_size=100M" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--source include/wait_until_connected_again.inc
+
+# check tables
+SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TOKUDB_FILE_MAP WHERE TABLE_SCHEMA='test' AND TABLE_DICTIONARY_NAME='main';
+OPTIMIZE TABLE test.t1;
+SELECT * FROM test.t1;
+SELECT * FROM test.t2;
+DROP TABLE test.t1;
+DROP TABLE test.t2;
+disconnect conn1;
+disconnect conn2;
+
+--echo =====2) crash and recover test=====
+connect (conn1, localhost, root,,);
+connection conn1;
+CREATE TABLE test.t2 (c1 INT, c2 INT, PRIMARY KEY(c1), CLUSTERING KEY(c2)) ENGINE=TOKUDB;
+
+--disable_query_log
+--let $i=1000
+while ($i)
+{
+  --eval INSERT INTO test.t2 VALUES($i, $i)
+  --dec $i
+}
+--enable_query_log
+
+--disable_query_log
+call mtr.add_suppression("Attempting backtrace");
+--enable_query_log
+
+# make mtr expect the "crash"
+--let $_server_id= `SELECT @@server_id`
```

Comment:
If this snippet stays in 5.6, just remove and use mysqld.1.expect below, this is not a generic snippet for arbitrary replication testcase server